src/region_growing.jl

```diff
@@ -0,0 +1,163 @@
+
+# An implementation of the Improved Seeded Region Growing Algorithm in Julia
+#
+# Citation:
+# Albert Mehnert, Paul Jackaway (1997), "An improved seeded region growing algorithm"
+# http://www.ee.bgu.ac.il/~itzik/IP5211/Other/Projects/P19_Seeded%20region%20growing.pdf
+
+function seeded_region_growing{CT<:Colorant, N}(img::AbstractArray{CT,N}, seeds::AbstractVector{Tuple{CartesianIndex{N},Int}},
+    kernel_dim::Vector{Int} = [3 for i in 1:N], diff_fn::Function = (c1,c2)->(sqrt(sum(abs2,wrapping_type(CT)(c1)-wrapping_type(CT)(c2)))))
+    length(kernel_dim) == N || error("Dimension count of image and kernel_dim do not match")
+    for dim in kernel_dim
+        dim > 0 || error("Dimensions of the kernel must be positive")
+        isodd(dim) || error("Dimensions of the kernel must be odd")
+    end
+    pt = CartesianIndex([floor(Int, dim/2) for dim in kernel_dim]...)
+    seeded_region_growing(img, seeds, ((c)->CartesianRange(c-pt,c+pt)), diff_fn)
+end
+
+function seeded_region_growing{CT<:Colorant, N}(img::AbstractArray{CT,N}, seeds::AbstractVector{Tuple{CartesianIndex{N},Int}},
+    neighbourhood::Function, diff_fn::Function = (c1,c2)->(sqrt(sum(abs2,wrapping_type(CT)(c1)-wrapping_type(CT)(c2)))))
+
+    # Check if labels are positive integers
+    for seed in seeds
+        seed[2] > 0 || error("Seed labels need to be positive integers!")
+    end
+
+    # Required data structures
+    result              =   similar(dims->fill(-1,dims), indices(img))              # Array to store labels
+    nhq                 =   Queue(CartesianIndex{N})                                # Neighbours holding queue
+    pq                  =   PriorityQueue(Queue{CartesianIndex{N}}, Float64)        # Priority Queue to hold the queues of same δ value
+    qdict               =   Dict{Float64, Queue{CartesianIndex{N}}}()               # A map to get a reference to queue using the δ value
+    labelsq             =   Queue(Int)                                              # Queue to hold labels
+    holdingq            =   Queue(CartesianIndex{N})                                # Queue to hold points corresponding to the labels in `labelsq`
+    region_means        =   Dict{Int, wrapping_type(CT)}()                          # A map containing (label, mean) pairs
+    region_pix_count    =   Dict{Int, Int}()                                        # A map containing (label, pixel_count) pairs
+    labels              =   Vector{Int}()                                           # A vector containing list of labels
```

Comment:
Really improves the readability to have this "table" (with comments), thanks for doing it.